api/v1alpha7/types.go

```diff
@@ -163,18 +174,42 @@ type RootVolume struct {
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 }
 
+// AdditionalBlockDevice is a block device to attach to the server.
 type AdditionalBlockDevice struct {
-	// Name of the Cinder volume in the context of a machine.
-	// It will be combined with the machine name to make the actual volume name.
+	// Type is the type of block device to create.
+	// This can be either "volume" or "local".
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=volume;local
+	// +unionDiscriminator
```

Comment:
@mdbooth please review the API now that we have unions.

Reply:
re: Union, I checked CAPZ (quite a reference to me) and they have one `unionDiscriminator` in their API and the `enum` is defined at the tpe level so for us that would be in `BlockDeviceType`. In OpenShift API we have set the enum differently to allow more granular values but I think in CAPO I'll follow what CAPZ did because it looks cleaner here.

Other than that I think the API is "final" for this PR and ready to be reviewed.